src/org/zaproxy/zap/extension/websocket/files/scripts/templates/websocketpassive/Passive WebSocket Scan Template.js

```diff
@@ -0,0 +1,81 @@
+// Passive scan rules should not send messages
+// Right click the script in the Scripts tree and select "enable"  or "disable"
+
+var PluginPassiveScanner = Java.type("org.zaproxy.zap.extension.websocket.pscan.WebSocketPluginPassiveScanner");
+
+OPCODE_CONTINUATION = 0x0;
+
+OPCODE_TEXT = 0x1;
+OPCODE_BINARY = 0x2;
+
+OPCODE_CLOSE = 0x8;
+OPCODE_PING = 0x9;
+OPCODE_PONG = 0xA;
+
+RISK_INFO 	= 0;
+RISK_LOW 	= 1;
+RISK_MEDIUM = 2;
+RISK_HIGH 	= 3;
+
+CONFIDENCE_LOW = 1;
+CONFIDENCE_MEDIUM = 2;
+CONFIDENCE_HIGH = 3;
+
+
+/**
+ * This function is Optional. By default allows all the message to pass at the scan phase.
+ * Return {@code true} to pass WebSocket message to scan function.
+ * @param msg WebSocket message (WebSocketMessage DTO)
+ */
+function applyScanToMessage(msg) {
+    if(msg.opcode == OPCODE_TEXT && !msg.isOutgoing){
+        return true;
+    }
+    return false;
+}
+
+/**
+ * Passively scans an WebSocket message. The scan function will be called for
+ * messages via ZAP. Script scans only the messages which is passed from
+ * from WebSocketPassiveScript#applyScanToMessage (defined bellow)
+ *
+ * @param alrt - the WebSocketPassiveScanAlert interface provides the raiseAlert method in order
+ *               to raise the appropriate alerts
+ * @param msg - the Websocket Message being scanned. This is an WebSocketMessage object.
+ *
+ * Some useful functions and fields of WebSocketMessageDTO:
+ * msg.channel        -> Channel of the message (WebSocketChannelDTO)
+ * msg.id             -> Unique ID of the message (int)
+ * msg.opcode         -> Opcode of the message (int) (Opcodes defined bellow)
+ * msg.readableOpcode -> Textual representation of opcode (String)
+ * msg.isOutgoing     -> Outgoing or incoming message (boolean)
+ * msg.getReadablePayload() -> Return readable representation of payload
+ *
+ * Some useful functions and fields of WebSocketChannelDTO:
+ * channel.id         -> Unique ID of the message (int)
+ * channel.host       -> Host of the WebSocket Server (String)
+ * channel.port       -> Port where this channel is connected at. Usually 80 or 443.
+ * channel.url        -> URL used in HTTP handshake.
+ */
+function scan(alrt,msg) {
+    // Test the request and/or response here
```

Comment:
`or` is fine :+1:  (no need for "and/or")

Reply:
http://kingthorin.blogspot.com/2007/09/writing-rant.html

Reply:
I forgot what :+1:

Reply:
No worries, it's kind of a finagly point... :grin: